cmd/kubeadm/app/cmd/config.go

```diff
@@ -206,8 +207,74 @@ func uploadConfiguration(client clientset.Interface, cfgPath string, defaultcfg
 	return uploadconfig.UploadConfiguration(internalcfg, client)
 }
 
-// NewCmdConfigListImages returns the "kubeadm images" command
-func NewCmdConfigListImages(out io.Writer) *cobra.Command {
+// NewCmdConfigImages returns the "config images" command
+func NewCmdConfigImages(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "images",
+		Short: "Interact with container images used by kubeadm.",
+		RunE:  cmdutil.SubCmdRunE("images"),
+	}
+	cmd.AddCommand(NewCmdConfigImagesList(out))
+	cmd.AddCommand(NewCmdConfigImagesPull())
+	return cmd
+}
+
+// NewCmdConfigImagesPull returns the `config images pull` command
+func NewCmdConfigImagesPull() *cobra.Command {
+	cfg := kubeadmapiv1alpha1.MasterConfiguration{}
+	kubeadmapiv1alpha1.SetDefaults_MasterConfiguration(&cfg)
+	var cfgPath, featureGatesString string
+	var err error
+	cmd := &cobra.Command{
+		Use:   "pull",
+		Short: "Pull images used by kubeadm.",
+		Run: func(_ *cobra.Command, _ []string) {
+			cfg.FeatureGates, err = features.NewFeatureGate(&features.InitFeatureGates, featureGatesString)
+			kubeadmutil.CheckErr(err)
+			internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, &cfg)
+			kubeadmutil.CheckErr(err)
+			puller, err := images.NewImagePuller(utilsexec.New(), internalcfg)
+			kubeadmutil.CheckErr(err)
+			imagesPull := NewImagesPull(puller, images.GetAllImages(internalcfg))
+			kubeadmutil.CheckErr(imagesPull.PullAll())
+		},
+	}
+	AddImagesCommonConfigFlags(cmd.PersistentFlags(), &cfg, &featureGatesString)
+	return cmd
+}
+
+// Puller is an interface for pulling images
+type Puller interface {
```

Comment:
I was thinking about this as if it were an internal interface (which I should have then named it `puller` instead) -- not really something for reuse, but I don't see any reason why it can't live in the images package.

The `ImagesPull` was not designed for use elsewhere, but looking at it a bit closer, the only thing I can see that is specific to this command is the `glog` call.

Would you be ok if I moved the Puller interface out to the `images` package but kept ImagesPull internal to this package for now?

Reply:
Ever if there is no real use case for reuse now, I would like to have the capability to pull a set of images implemented in the `app/images` package instead of in the UX

> ..., the only thing I can see that is specific to this command is the `glog` call.
I don't see this as a constraint, e.g. there are already several glog calls in phases